deps/media-playback/media-playback/media.c

```diff
@@ -653,93 +637,349 @@ static bool init_avformat(mp_media_t *m)
+
+	if (!mp_media_reset(m, oldstate))
 		return false;
+
+	mp_media_push_message(m, mstate_playing);
+
+	return true;
+}
+
+/*Executes when kill event occurs.*/
+static inline bool mp_media_killing(mp_media_t *m)
+{
+	m->mp_state = mstate_killing;
+	m->killing = true;
+	return true;
+}
+
+/*Render the frame.*/
+static inline bool mp_media_rendering(mp_media_t *m, bool timeout)
+{
+	if (!timeout && m->mp_state != mstate_stopped) {
+		mp_render_frame(m);
 	}
 
-	for (;;) {
-		bool reset, kill, is_active, seek, pause, reset_time;
-		int64_t seek_pos;
-		bool timeout = false;
+	return true;
+}
 
-		pthread_mutex_lock(&m->mutex);
-		is_active = m->active;
-		pthread_mutex_unlock(&m->mutex);
+/*Executes after leaving a state. Keep in mind that this won't execute
+  until AFTER the entry method of the new state. Therefore, if cleanup
+  is needed before changing states, put that logic in the entry
+  function of the new state.*/
+static inline void mp_media_change_state(mp_media_t *m,
+					 enum mp_media_state oldstate)
+{
+	switch (oldstate) {
+	case mstate_uninit:
 
-		if (!is_active) {
-			if (os_sem_wait(m->sem) < 0)
-				return false;
-		} else {
-			timeout = mp_media_sleepto(m);
-		}
+		break;
+	case mstate_playing:
 
-		pthread_mutex_lock(&m->mutex);
+		break;
+	case mstate_paused:
+		/*reset time stamp to prevent fast forward*/
+		reset_ts(m);
+		break;
+	case mstate_resume:
 
-		reset = m->reset;
-		kill = m->kill;
-		m->reset = false;
-		m->kill = false;
+		break;
+	case mstate_seeking:
 
-		pause = m->pause;
-		seek_pos = m->seek_pos;
-		seek = m->seek;
-		reset_time = m->reset_ts;
-		m->seek = false;
-		m->reset_ts = false;
+		break;
+	case mstate_resetting:
 
-		pthread_mutex_unlock(&m->mutex);
+		break;
+	case mstate_setloop:
 
-		if (kill) {
-			break;
-		}
-		if (reset) {
-			mp_media_reset(m);
-			continue;
-		}
+		break;
+	case mstate_setreconnect:
 
-		if (seek) {
-			m->seek_next_ts = true;
-			seek_to(m, seek_pos);
-			continue;
-		}
+		break;
+	case mstate_stopping:
 
-		if (reset_time) {
-			reset_ts(m);
-			continue;
-		}
+		break;
+	case mstate_stopped:
 
-		if (pause)
-			continue;
+		break;
+	case mstate_killing:
 
-		/* frames are ready */
-		if (is_active && !timeout) {
-			if (m->has_video)
-				mp_media_next_video(m, false);
-			if (m->has_audio)
-				mp_media_next_audio(m);
+		break;
+	}
+}
 
-			if (!mp_media_prepare_frames(m))
-				return false;
-			if (mp_media_eof(m))
-				continue;
+/*Executes the top message in the queue.*/
+static bool mp_media_pop_message(mp_media_t *m)
+{
+	bool result = true;
+
+	mp_media_queue_node_t *front = m->mp_queue.first;
+	/*Only execute if a state change actually occurs. If an
+	  action needs to be called within the state, add a new
+	  state that executes the action then moves back.*/
+	if (m->mp_state != front->message.state) {
+		result = front->message.state_handler(m);
+	}
+
+	pthread_mutex_lock(&m->mutex);
+
+	if (m->mp_queue.first->next != NULL)
```

Comment:
Don't mix no-curly-braces blocks with curly-braces blocks in the same statement.

See [Section 3 of the Linux kernel coding style](https://github.com/torvalds/linux/blob/2324d50d051ec0f14a548e78554fb02513d6dcef/Documentation/process/coding-style.rst#3-placing-braces-and-spaces) which is linked from [our Contributing Guidelines](https://github.com/obsproject/obs-studio/blob/7031e2ec3a6ab2f97e7d8f50640d16b306f01e59/CONTRIBUTING.rst).